RFS/src/main/java/com/rfs/common/RestClient.java

```diff
@@ -48,7 +51,18 @@ public class RestClient {
     private static final String JSON_CONTENT_TYPE = "application/json";
 
     public RestClient(ConnectionContext connectionContext) {
-        this(connectionContext, HttpClient.create());
+        this(connectionContext, 0);
+    }
+
+    /**
+     * @param maxConnections If &gt; 0, an HttpClient will be created with a provider
+     *                       that uses this value for maxConnections.  Otherwise, a client
+     *                       will be created with default values provided by Reactor.
+     */
+    public RestClient(ConnectionContext connectionContext, int maxConnections) {
```

Comment:
Seems like the connection context should include the max connection

Reply:
One is a pool configuration and the other is necessary for a single client.

Reply:
I'm don't understand the distinction, what is the case for allowing them to be set separately?  Is this an issue with rfs use of connection context vs how the proxy/replayer use it?